content/en/docs/instrumentation/python/cookbook.md

```diff
@@ -94,7 +107,15 @@ span_context = SpanContext(
 )
 ctx = trace.set_span_in_context(NonRecordingSpan(span_context))
 
-with tracer.start_as_current_span("child", context=ctx) as span:
+# Now there are a few ways to make use of the trace context.
+
+# You can pass the context object when starting a span.
+with tracer.start_as_current_span('child', context=ctx) as span:
+    span.set_attribute('primes', [2, 3, 5, 7])
+
+# Or you can make it the current context, and then the next span will pick it up.
+context.attach(ctx)
```

Comment:
The best practice here would be wrapping this in a `try` and detaching in the `finally`. I believe it would look like

```py
try:
  token = context.attach(ctx)
  ...
finally:
  context.detach(token)
```

otherwise the context can leak

Reply:
Thanks, updated. I put the `attach` before the `try` since if that fails you won't have a `token` to detach with.